### PR TITLE
Update operator-sdk version to v1.24

### PR DESCRIPTION
**Pull request by halim-lee**
**What this PR does / why we need it?**:

- Update operator-sdk version to v1.24
- Update kubernetes APIs to 1.24.2

**Does this PR introduce a user-facing change?**

- [ ] User guide
- [ ] `CHANGELOG.md`

**Which issue(s) this PR fixes**:

Related: https://github.com/OpenLiberty/open-liberty-operator/issues/376

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -176,43 +155,6 @@ kustomize: $(KUSTOMIZE) ## Download kustomize locally if necessary.
...
-go mod init tmp ;\
-echo "Downloading $(2)" ;\
-GOBIN=$(PROJECT_DIR)/bin go get $(2) ;\
-rm -rf $$TMP_DIR ;\
-}
-endef
-
-.PHONY: bundle
-bundle: manifests setup kustomize ## Generate bundle manifests and metadata, then validate generated files.
-	scripts/update-sample.sh
```
Comment (leochr) [inline Makefile:195]: I believe this was removed accidentally. This line was added by PR https://github.com/OpenLiberty/open-liberty-operator/pull/375/files and should be retained.

**Review by leochr — COMMENTED** [review]
@halim-lee Thank you for the PR. Looks good. Just one minor comment

File: Makefile (hunk 1 above)
Comment (leochr) [reply Makefile:195]: @halim-lee I referred to this line in particular: `scripts/update-sample.sh` which is not present in the make target `bundle` ([here](https://github.com/OpenLiberty/open-liberty-operator/blob/58206329f707d74385c3b1357d3d8c4de484e206/Makefile#L202)). Could it be that it's not pushed to remote?

Comment (halim-lee) [reply Makefile:195]: Oh no - it's added back in!

**Review by leochr — APPROVED** [review]
@halim-lee Thanks for the update. Looks good.

**Pull request merged**